scipy/optimize/_differentialevolution.py

```diff
@@ -1116,6 +1153,29 @@ def solve(self):
 
         return DE_result
 
+    def _result(self, **kwds):
+        # form an intermediate OptimizeResult
+        nit = kwds.get('nit', None)
+        message = kwds.get('message', None)
+        warning_flag = kwds.get('warning_flag', False)
+        result = OptimizeResult(
+            x=self.x,
+            fun=self.population_energies[0],
+            nfev=self._nfev,
+            nit=nit,
+            message=message,
+            success=(warning_flag is not True),
+            population=self._scale_parameters(self.population)
+        )
+        if self._wrapped_constraints:
```

Comment:
Are these attributes and when they are available documented? Not necessarily for this PR, but thought I'd check because I don't see them in https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.differential_evolution.html#scipy.optimize.differential_evolution.

Reply:
The `DifferentialEvolutionSolver` class is not public, it's called by the public `differential_evolution` function which does not have any of those attributes.

So attributes of a private class don't necessarily need to be documented.

Reply:
Oops, bad line reference - I meant the attributes of the return object.  For example, `population_energies` is not defined.

Reply:
I've just added `population_energies` and `population` to the docstring.

Reply:
Cool. It was unrelated to this issue, so it would have been fine to do it later, but good to have it done.